### ISSUE TITLE
[Inductor][CPU] as_strided addcmul_ lowers through a view-sized temporary

### ISSUE THREAD

**Issue by ruddyscent** [pytorch/pytorch#195437]
## 🐛 Describe the bug

On CPU, Inductor materializes an `as_strided` target-view buffer for an in-place `addcmul_`, then copies that buffer back into the original tensor.

The target below is a non-overlapping subview of `hx`; `hx`, `ey`, and `inv_hx` are independently allocated.  Eager and compiled results agree, so this appears to be a memory/performance issue rather than a correctness issue.

This is related to #195330's compact indexed-workspace allocation, but it has no runtime indices or caller-provided workspace: the staging allocation here comes from lowering a mutation through an `as_strided` destination view.

## Minimal reproduction

Run this in a fresh Python process:

```python
import torch

torch.set_num_threads(1)
torch.set_num_interop_threads(1)
torch.manual_seed(0)


def stencil(hx, ey, inv_hx):
    target = torch.as_strided(
        hx,
        size=(18, 15, 12, 2),
        stride=hx.stride(),
        storage_offset=28,
    )
    target.addcmul_(
        inv_hx[:, 1:, 1:],
        (ey[:-1, :, 1:] - ey[:-1, :, :-1]) * 3.0,
        value=0.2545496022389349,
    )


hx0 = torch.randn((18, 16, 13, 2), dtype=torch.float64)
ey = torch.randn((19, 15, 13, 2), dtype=torch.float64)
inv_hx = torch.randn((18, 16, 13, 1), dtype=torch.float64)

eager_hx = hx0.clone()
compiled_hx = hx0.clone()
stencil(eager_hx, ey, inv_hx)

compiled = torch.compile(
    stencil,
    backend="inductor",
    fullgraph=True,
    dynamic=False,
)
compiled(compiled_hx, ey, inv_hx)  # warm-up
torch.testing.assert_close(compiled_hx, eager_hx, rtol=1e-12, atol=1e-12)
```

To inspect the generated wrapper, run with a fresh cache and search it for `empty_strided_cpu`:

```bash
cache=$(mktemp -d)
TORCHINDUCTOR_CACHE_DIR="$cache" python repro.py
rg -n "empty_strided_cpu|cpp_fused.*copy" "$cache"
```

## Actual behavior

The generated CPU wrapper contains this allocation and then passes it to a fused kernel that also receives the original `hx` storage:

```python
buf1 = empty_strided_cpu(
    (18, 15, 12, 2),
    (360, 24, 2, 1),
    torch.float64,
)
```

`buf1` has `18 * 15 * 12 * 2 = 6,480` `float64` elements, or 51,840 bytes. That is 86.5% of the 59,904-byte base `hx` tensor.  The generated code first writes the updated view values into this buffer and then copies them back to the `as_strided` location in `hx`.

On the warmed compiled call, the standard CPU profiler reports no raw or operator-level allocation event for this buffer, even though the wrapper explicitly constructs it.  The same profiler does record a 51,840-byte `aten::empty_strided` allocation for a direct/eager version, so generated-code inspection was necessary for the compiled path.

I reviewed the results and agree that the current staging path is faster than direct writes.

<details open>
<summary><strong>AI-assisted microbenchmark supplement</strong></summary>

> **AI assistance disclosure:** Codex prepared and ran the microbenchmark, inspected the generated wrapper, kernel, and assembly, and drafted the analysis below. The submitting human reviewed the exact commands and results before authorizing this update.
>
> ### Microbenchmark
>
> I measured the exact reproducer with two compiler variants:
>
> - **Current staging path:** the unmodified generated wrapper and kernel.
> - **Diagnostic direct store:** a compile-time-only control that kept mutation bookkeeping, but for this exact graph skipped eager realization of 10 registered readers across two mutation marks and skipped the mutation source `realize_hint()` once. This removed the 51,840-byte `empty_strided_cpu` allocation and changed the generated kernel from five arguments and two loop nests to four arguments and one direct-store loop. This control is not a proposed general lowering change.
>
> Each result is the median [Q1, Q3] of 31 randomized paired samples pinned to CPU 0 with intra-op, inter-op, OpenMP, MKL, and OpenBLAS threads set to 1. Each side ran for about 100 ms per sample after 200 warm-up calls; compilation and input resets were outside the timed regions.
>
> | Timed level | Current staging path | Diagnostic direct store | Paired direct-store change |
> | --- | ---: | ---: | ---: |
> | Warmed `torch.compile` call (2,177 calls/sample) | 46.328 us [46.158, 46.602] | 52.517 us [52.074, 52.738] | +6.160 us [5.623, 6.614], 1.133x |
> | Generated `Runner.call` (10,880 calls/sample) | 9.475 us [9.406, 9.666] | 18.972 us [18.760, 19.196] | +9.361 us [9.162, 9.620], 1.993x |
> | Raw generated kernel (13,689 calls/sample) | 7.476 us [7.361, 7.630] | 18.429 us [18.070, 18.626] | +10.900 us [10.682, 11.080], 2.460x |
>
> I also isolated the steady-state allocation component while keeping the current five-argument kernel unchanged. Allocating a fresh 51,840-byte stage on every call took 8.411 us [8.329, 8.513], versus 7.519 us [7.462, 7.688] with one preallocated stage reused across calls (13,649 calls/sample). The paired allocation cost was +0.880 us [0.798, 0.990], or 1.117x.
>
> Both compiler variants matched eager at `rtol=1e-12, atol=1e-12`. The current and diagnostic variants remained bitwise equal after 1,000 repeated generated-wrapper calls and 1,000 repeated raw-kernel calls. Code inspection confirmed one view-sized allocation became zero and kernel arity changed from five to four.
>
> On this Skylake CPU, the diagnostic direct path uses two-element masked loads and stores on the same strided destination, and GCC fully unrolls the 12-iteration inner loop. The current path instead writes the arithmetic result to a compact stage and copies it with full-width eight-double loads and stores. Therefore this measurement does **not** support claiming that the current staging path is a latency regression on this reproducer: the warmed allocation itself has a measurable approximately 0.88 us cost, but the naive direct-store code generation is slower by a larger amount. Any optimization that removes the staging buffer needs to preserve or improve the current vectorization and should be benchmarked independently.
>
> Environment: Intel Core i7-6700, Linux 6.8.0-138-generic, GCC 13.3.0, Python 3.12.3, PyTorch source checkout `7bdc7ec2a3cd822a2cb22f4af0032c66496387f6`, and loaded native build `2.15.0a0+gitb3632e5` (`b3632e56519f39da86e50fb0aa27235d626e8306`). The relevant Inductor Python sources came from the checkout; the native-extension revision mismatch is a limitation of this measurement.

</details>

## Expected behavior

For this disjoint-input, non-overlapping-view case, Inductor should ideally update the `hx` view directly, or otherwise avoid allocating a view-sized staging tensor on every invocation.  This looks like a missed alias-aware/reinplacing optimization for `as_strided` mutation targets.

## Environment

- PyTorch nightly: `2.15.0.dev20260829+cpu`
- PyTorch git revision: `332a69317e22b105a867838624a87984e05021e2`
- CPU-only, `float64`, `backend="inductor"`, `fullgraph=True`, `dynamic=False`
